src/dodal/devices/i18/sim_detector.py

```diff
@@ -0,0 +1,18 @@
+from ophyd_async.epics.motor import Motor
+from ophyd_async.sim.demo import PatternGenerator
+
+
+class SimDetector:
+    def __init__(self, name: str, motor: Motor, motor_field: str):
+        self.name = name
+        self.motor = motor
+        self.motor_field = motor_field
+        self.pattern_generator = PatternGenerator(
+            saturation_exposure_time=0.1, detector_height=100, detector_width=100
+        )
```

Comment:
Is there a reason you're redefining a SimDetector here rather than using the one in ophyd-async that uses the PatternGenerator?

Reply:
The motor field isn't being used at all- is the PatternGenrator meant to read the motor and change its output depending on the value of the motor?

Reply:
I am not sure of the details - what do you think @iain-hall ?

Reply:
I'm not familiar with PatternGenerator - could be useful. Any simulated detectors would be generally be used for offline testing rather than on the beamline, and we have one already for the diode used in the undulator gap/lookup table generator scan. Maybe we should only have real beamline devices in beamlines/i18.py, and add simulated devices as needed for offline testing?

Reply:
the motor fields could be used in a plan with dot notation

Reply:
I think you should lose the internal motor- you don't want to write your plans with the assumption that your Detector will have an internal motor, you want to write them to be generic enough to use StandardDetector: this will allow you to re-use the plans you're writing for your simulated detector. If you're not going to be using the position of the motor as an input to your PatternGenerator, I would remove it. If you are, I would make it an internal/private field.